### PR TITLE
Merge paradox/scala/mailboxes.md and java/mailboxes.md

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #23052

## Rendered HTML:

- https://richard-akka-doc.firebaseapp.com/scala/mailboxes.html
- https://richard-akka-doc.firebaseapp.com/java/mailboxes.html

## I split this in two commits:

- **1st commit:** Identical contents in scala/mailboxes.md and java/mailboxes.md -> **this is to take the diff both in scala/mailboxes.md and java/mailboxes.md**

  - [scala/mailboxes.md  (prev) vs. scala/mailboxes.md (after)](https://github.com/richard-imaoka/akka/commit/7f047fb6a57e1ea0abf000985c7f60491ee30512#diff-1f388ac46dff4266e655ad6dd5917f5f)
  - [java/mailboxes.md (prev) vs. java/mailboxes.md (after)](https://github.com/richard-imaoka/akka/commit/7f047fb6a57e1ea0abf000985c7f60491ee30512#diff-4af2c5329c5060130ed970b2ba48d73b)

- **2nd commit:** Replace java/mailboxes.md with a symlink to scala/mailboxes.md

### CODE REVIEW

File: akka-docs/src/main/paradox/scala/mailboxes.md [hunk 1]
```diff
@@ -214,17 +238,36 @@ It can be configured like this:
...
 An example is worth a thousand quacks:
 
-@@snip [MyUnboundedMailbox.scala]($code$/scala/docs/dispatcher/MyUnboundedMailbox.scala) { #mailbox-implementation-example }
+Scala
+:   @@snip [MyUnboundedMailbox.scala]($code$/scala/docs/dispatcher/MyUnboundedMailbox.scala) { #mailbox-marker-interface }
+
+Java
+:   @@snip [MyUnboundedMessageQueueSemantics.java]($code$/java/jdocs/dispatcher/MyUnboundedMessageQueueSemantics.java) { #mailbox-marker-interface }
+
+
```
Comment (richard-imaoka) [inline akka-docs/src/main/paradox/scala/mailboxes.md]: Split into `#mailbox-marker-interface` and `#mailbox-marker-interface` to be consistent in Scala and Java, along with changes in `MyUnboundedMessageQueueSemantics.java` and `MyUnboundedMailbox.scala`

image

File: akka-docs/src/test/java/jdocs/dispatcher/MyUnboundedMessageQueueSemantics.java [hunk 2]
```diff
@@ -4,8 +4,8 @@
 
 package jdocs.dispatcher;
 
-//#mailbox-implementation-example
+//#mailbox-marker-interface
 // Marker interface used for mailbox requirements mapping
 public interface MyUnboundedMessageQueueSemantics {
 }
-//#mailbox-implementation-example
+//#mailbox-marker-interface
```
Comment (richard-imaoka) [inline akka-docs/src/test/java/jdocs/dispatcher/MyUnboundedMessageQueueSemantics.java]: Refer to [this comment](https://github.com/akka/akka/pull/23191/files#r122614439).

File: akka-docs/src/test/scala/docs/dispatcher/MyUnboundedMailbox.scala [hunk 3]
```diff
@@ -15,9 +15,6 @@ import com.typesafe.config.Config
 import java.util.concurrent.ConcurrentLinkedQueue
 import scala.Option
 
-// Marker trait used for mailbox requirements mapping
-trait MyUnboundedMessageQueueSemantics
-
```
Comment (richard-imaoka) [inline akka-docs/src/test/scala/docs/dispatcher/MyUnboundedMailbox.scala]: Refer to [this comment](https://github.com/akka/akka/pull/23191/files#r122614439).

**Review by raboof — APPROVED** [review]
:+1:

**Pull request merged**